### PR TITLE
test: migrate password tests to hono

### PR DESCRIPTION
**Pull request by danielgent**
PR #2 of the latest round of wrangler migrations

### CODE REVIEW

File: test/integration/flows/password.spec.ts [hunk 1]
```diff
@@ -1,77 +1,82 @@
...
-            password: "password",
-          }),
         },
       );
+      // so this is correct
+      expect(await response.text()).toBe("Client not found");
 
-      expect(response.status).toBe(400);
+      // but this is giving a 404...
+      // expect(response.status).toBe(400);
```
Comment (danielgent) [inline test/integration/flows/password.spec.ts:35]: :eyes: this needs looking at...

We're getting the correct error message... but the wrong status code :thinking:

File: test/integration/flows/password.spec.ts
Comment (danielgent) [inline test/integration/flows/password.spec.ts:1]: so much nicer when we get a diff :smile:

File: test/integration/flows/password.spec.ts [hunk 2]
```diff
@@ -1,77 +1,82 @@
...
-  });
-
   describe("Register password", () => {
     it("should return a 400 if an invalid client is passed", async () => {
-      const response = await worker.fetch(
-        "/invalidClientId/dbconnection/register",
+      const env = await getEnv();
+      const client = testClient(tsoaApp, env);
+
+      const typesDoNotWorkWithThisSetup___PARAMS = {
```
Comment (danielgent) [inline test/integration/flows/password.spec.ts:14]: :smiling_face_with_tear:  would be great to fix this somehow

File: test/integration/flows/password.spec.ts [hunk 3]
```diff
@@ -1,77 +1,80 @@
...
-          body: JSON.stringify({
-            email: "test@example.com",
-            password: "password",
-          }),
         },
       );
+      expect(await response.text()).toBe("Client not found");
 
-      expect(response.status).toBe(400);
+      expect(response.status).toBe(404);
```
Comment (danielgent) [inline test/integration/flows/password.spec.ts:33]: Note this change.  Now we never reach our route code to throw a 400

```
  const client = await getClient(ctx.env, clientId);

    if (!client) {
      throw new HTTPException(400, { message: "Client not found" });
    }
```    


Instead the Kysely adapter does

```
export function createClientsAdapter(db: Kysely<Database>) {
  return {
    get: async (applicationId: string) => {
      const application = await db
        .selectFrom("applications")
        .selectAll()
        .where("id", "=", applicationId)
        .executeTakeFirst();

      if (!application) {
        throw new HTTPException(404, { message: "Client not found" });
      }
```

File: test/integration/flows/password.spec.ts [hunk 4]
```diff
@@ -1,77 +1,80 @@
...
-          client_id: "clientId",
-          credential_type: "http://auth0.com/oauth/grant-type/password-realm",
-          realm: "Username-Password-Authentication",
-          password,
-          username: "password-login-test@example.com",
-        }),
       });
 
-      expect(loginResponse.status).toBe(200);
+      expect(createUserResponse.status).toBe(201);
```
Comment (danielgent) [inline test/integration/flows/password.spec.ts:57]: the diff has got a bit out of sync here :smiley:

File: test/integration/flows/password.spec.ts [hunk 5]
```diff
@@ -80,60 +83,52 @@ describe("password-flow", () => {
         redirect_uri: "http://login.example.com",
         state: "state",
         realm: "Username-Password-Authentication",
-      });
-
+      };
       // Trade the ticket for token
```
Comment (danielgent) [inline test/integration/flows/password.spec.ts:87]: the comments snap the diff back... I hadn't realise this advantage

File: test/integration/flows/password.spec.ts (hunk 2 above)
Comment (markusahlstrand) [reply test/integration/flows/password.spec.ts:14]: Yeah, lets see if we can make a PR to the hono project..

File: test/integration/flows/password.spec.ts (hunk 3 above)
Comment (markusahlstrand) [reply test/integration/flows/password.spec.ts:33]: I wonder what the correct response is? Guessing that we should catch this exception and throw a 400 instead? Let's check how the auth0 response looks like

**Review by markusahlstrand — APPROVED** [review]
Looks good. Nice with the move of the file.. I would have just delete the old file a lost the diff

**Comment by danielgent** [comment]
> Looks good. Nice with the move of the file.. I would have just delete the old file a lost the diff

Sadly `git mv` doesn't seem to make a whole lot of difference to github

Github will work out if the file has moved it seems (as it has on your PRs).  Maybe it's just implies to github that it should show a diff (as it seems more likely) but it's not guaranteed

**Pull request merged**